parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -219,34 +171,27 @@ pub mod pallet {
 				Error::<T>::PayloadTooLarge,
 			);
 
-			let next_id = <NextId<T>>::get();
-			if next_id.checked_add(1).is_none() {
-				return Err(Error::<T>::Overflow.into())
-			}
+			let nonce = <Nonce<T>>::get(who);
+			let next_nonce = nonce.saturating_add(1);
+			ensure!(next_nonce != nonce, Error::<T>::Overflow);
```

Comment:
seems the `ensure` logic is not correct here, `next_nonce` will be zero when overflow, use this instead
`nonce.checked_add(1).ok_or(Error::<T>::Overflow)?`

Reply:
Saturating_add doesn't give 0 on an overflow:
```
let x = 255u8.saturating_add(1);
println!("{:?}", x); // prints "255"
```

That said, we don't actually have a use for the saturating behaviour. We're returning an error anyway, so `checked_add` is a better choice.